modules/ExperimentBrowser/src/client/ExperimentBrowser.coffee

```diff
@@ -610,114 +609,23 @@ class ExperimentBrowserController extends Backbone.View
-			# Preparatory Logic to Get Code to Pass to Generate Link Route 
-			if @experimentController.model.get('lsLabels') not instanceof LabelList
-				@experimentController.model.set 'lsLabels',  new LabelList @experimentController.model.get('lsLabels')
-			subclass = @experimentController.model.get('subclass')
-			if window.conf.entity?.saveInitialsCorpName? and window.conf.entity.saveInitialsCorpName is true
-				if @experimentController.model.get('lsLabels').getLabelByTypeAndKind("corpName", subclass + ' corpName').length > 0
-					code = @experimentController.model.get('lsLabels').getLabelByTypeAndKind('corpName', subclass + ' corpName')[0].get('labelText')
-				else if @experimentController.model.get('lsKind') is "study" and @experimentController.model.get('lsLabels').getLabelByTypeAndKind('id', 'study id').length > 0
-					code = @experimentController.model.get('lsLabels').getLabelByTypeAndKind('id', 'study id')[0].get('labelText')
-				else
-					code = @experimentController.model.get("codeName")
-			else if @experimentController.model.get('lsKind') is "study" and @experimentController.model.get('lsLabels').getLabelByTypeAndKind('id', 'study id').length > 0
-				code = @experimentController.model.get('lsLabels').getLabelByTypeAndKind('id', 'study id')[0].get('labelText')
-			else
-				code = @experimentController.model.get('codeName')
-			# Add Generating Link Loading Mask 
-			@$('.bv_generatingLink').show()
-			# Call to Route to Get URL 
-			$.ajax
-				type: 'GET'
-				url: "/api/getLinkExptQueryTool?experiment=#{code}"
-				success: (response) => 
-					# Take Away Generating Progress Mask 
-					@$('.bv_generatingLink').hide()
-					window.open(response,'_blank')
-				error: (err) =>
-					# Take Away Generating Progress Mask 
-					@$('.bv_generatingLink').hide()
-					console.log err
-				datatype: 'json'
-
-	handleGetLinkQueryToolClicked: => 
-			# Preparatory Logic to Get Code to Pass to Generate Link Route 
-			if @experimentController.model.get('lsLabels') not instanceof LabelList
-				@experimentController.model.set 'lsLabels',  new LabelList @experimentController.model.get('lsLabels')
-			subclass = @experimentController.model.get('subclass')
-			if window.conf.entity?.saveInitialsCorpName? and window.conf.entity.saveInitialsCorpName is true
-				if @experimentController.model.get('lsLabels').getLabelByTypeAndKind("corpName", subclass + ' corpName').length > 0
-					code = @experimentController.model.get('lsLabels').getLabelByTypeAndKind('corpName', subclass + ' corpName')[0].get('labelText')
-				else if @experimentController.model.get('lsKind') is "study" and @experimentController.model.get('lsLabels').getLabelByTypeAndKind('id', 'study id').length > 0
-					code = @experimentController.model.get('lsLabels').getLabelByTypeAndKind('id', 'study id')[0].get('labelText')
-				else
-					code = @experimentController.model.get("codeName")
+
+	getExperimentCode: => 
```

Comment:
@dalejerikson  I just remembered we had talked about refactoring a few other places in ExperimentBrowser to use this utility function. Can you add that to this PR? It feels like the right time to clean that up.

Reply:
Yep, will do!

Reply:
Completed